src/Repl/Repl.tsx

```diff
@@ -0,0 +1,317 @@
+    if(decoded["boards"] && decoded["boards"] !== "") {
+      res = decodeValue(decoded["boards"]) + res
+    }
+
+    return res
+  }
+
+
+  /*
+   * handleTypeError
+   * Handles reporting a BoGL type error
+   *
+   * decoded: The decoded response we can use
+   */
+  const handleTypeError = (decoded) => {
+    let res = decoded["contents"]["message"]
+    if(inputMode === true) {
+      popFromInputBuffer();
+      res += "\nYour last input was discarded.";
+    }
+    return res
+  }
+
+
+  /*
+   * handleLanguageError
+   * Handles reporting a BoGL parse error
+   *
+   * decoded: The decoded response we can use
+   */
+  const handleLanguageError = (decoded) => {
+    let res = "Language Error: "+decoded["contents"][3];
+    if(inputMode === true) {
+      popFromInputBuffer();
+      res += "\nYour last input was discarded.";
+    }
+    return res
+  }
+
+
+  /*
+   * handleRuntimeError
+   * Handles reporting a BoGL runtime error
+   *
+   * decoded: The decoded response we can use
+   */
+  const handleRuntimeError = (decoded) => {
+    // Runtime Error
+    let res = "Runtime Error: "+decoded["contents"];
+    if(inputMode === true) {
+      popFromInputBuffer();
+      res += "\nYour last input was discarded.";
+    }
+    return res
+  }
+
+
+  /*
+   * handleValue
+   * Handles reporting a BoGL Value
+   *
+   * decoded: The decoded response we can use
+   */
+  const handleValue = (decoded) => {
+
+    let res = "";
+
+    // decode the value in the response
+    res += decodeValue(decoded)
+
+    // consider where we should report the type of the expression
+    if(shouldReportExprType) {
+      res += decodeExprType(decoded["type"]);
+      shouldReportExprType = false;
+    }
+
+    // lastly, report whether we are out of input mode
+    if(inputMode) {
+      // exit input mode
+      res += "\n\n 🤖 BoGL Says: Done reading input. \n";
+      exitInputMode();
+    }
+
+    return res;
+
+  }
+
+
+  /*
+   * handleResponse
+   * Handles the response from the bogl API w/ the replExpression that produced it
+   *
+   * replExpression : Expr used to produce this response
+   * responses :      Encoded reponse from the API
+   */
+  const handleResponse = (replExpression: string, responses: any) => {
+    // get recent decoded response
+    let decoded = decodeResponse(responses)
+
+    // Check for exception cases
+    if(decoded["category"] === "SpielPrompt") {
+      // Prompt for input
+      return handlePrompt(replExpression, decoded)
+
+    } else if(decoded["category"] === "SpielTypeError") {
+      // Type error
+      return handleTypeError(decoded)
+
+    } else if(decoded["category"] === "SpielParseError") {
+      // Parse Error
+      return handleLanguageError(decoded)
+
+    } else if(decoded["category"] === "SpielRuntimeError") {
+      // Runtime Error
+      return handleRuntimeError(decoded)
+
+    } else {
+      // Regular BoGL value
+      return handleValue(decoded)
```

Comment:
Looking back this should be changed a bit. The category may not be a SpielValue in rare cases, but it should be accounted for in some fashion